lib/Command/Announce.php

```diff
@@ -0,0 +1,197 @@
+		$this->logger = $logger;
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('announcementcenter:announce')
+			->setDescription('Create an announcement')
+			->addArgument(
+				'user',
+				InputArgument::REQUIRED,
+				'User who creates the announcement',
+			)
+			->addArgument(
+				'subject',
+				InputArgument::REQUIRED,
+				'Subject of the announcement',
+			)
+			->addArgument(
+				'message',
+				InputArgument::REQUIRED,
+				'Message of the announcement (supports markdown)',
+			)
+			->addOption(
+				'activities',
+				null,
+				InputOption::VALUE_NONE,
+				'Get notified over activities',
+			)
+			->addOption(
+				'notifications',
+				null,
+				InputOption::VALUE_NONE,
+				'Get notified over nextclouds notifications',
+			)
+			->addOption(
+				'emails',
+				null,
+				InputOption::VALUE_NONE,
+				'Notify users over email',
+			)
+			->addOption(
+				'comments',
+				null,
+				InputOption::VALUE_NONE,
+				'Allow comments',
+			)
+			->addOption(
+				'schedule-time',
+				's',
+				InputOption::VALUE_OPTIONAL,
+				'Publishing time of the announcement (see php strtotime)',
+				null,
+			)
+			->addOption(
+				'delete-time',
+				'd',
+				InputOption::VALUE_OPTIONAL,
+				'Deletion time of the announcement (see php strtotime)',
+				null,
+			)
+			->addOption(
+				'group',
+				'g',
+				InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+				'Group to set send announcement to (default "everyone", multiple allowed)',
+				['everyone'],
+			);
+	}
+
+	private function plainifyMessage(string $message) {
+		# TODO use Parsedown or Markdownify here
+		return $message;
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		// required
+		$user = $input->getArgument('user');
+		if (!$this->userManager->userExists($user)) {
+			throw new \InvalidArgumentException("User <$user> in unknown.");
+		}
+		$subject = $input->getArgument('subject');
+		$message = $input->getArgument('message');
+
+		// options
+		$groups = $input->getOption('group');
+
+		// notification types
+		$activities = $input->getOption('activities');
+		$notifications = $input->getOption('notifications');
+		$emails = $input->getOption('emails');
+		$comments = $input->getOption('comments');
+
+		// times
+		$scheduleTime = $this->parseTimestamp($input->getOption('schedule-time'));
+		$deleteTime = $this->parseTimestamp($input->getOption('delete-time'));
+
+		// validation
+		if ($scheduleTime && $deleteTime && $deleteTime < $scheduleTime) {
+			throw new \InvalidArgumentException("Publishing time is after deletion time");
+		}
+
+		$plainMessage = $this->plainifyMessage($message);
+		$notificationOptions = $this->notificationType->setNotificationTypes($activities, $notifications, $emails);
+
+		if (!$notificationOptions) {
+			throw new \InvalidArgumentException("You didn't set any notification option, please set 'activities', 'notifications' or 'emails'");
```

Comment:
This is not required in the web?

Reply:
Apprently not, your message just gets ignored: https://github.com/nextcloud/announcementcenter/blob/92c90218fdc7fa3db938ddeb48c50bcb882d1d37/lib/Manager.php#L149

Reply:
Well it's still posted and can be read from the announcementcenter page

Reply:
Oh! This is how this works, okay I'll remove this restriction